EIPS/eip-7743.md

```diff
@@ -0,0 +1,120 @@
+---
+eip: 7743
+title: Account Recovery by OIDC
+description: A new model to recovery account with OIDC ZK verification
+author: Shu Dong (@dongshu2013) <shu@hexlink.io>, Yudao Yan <dean@dauth.network>, Song Z <s@misfit.id>, Kai Chen <kai@dauth.network>
+discussions-to: https://ethereum-magicians.org/t/eip-7743-recoverable-account-with-oidc-zk-verification/15862
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-09-20
+---
+
+## Abstract
+
+Account Abstraction facilitates new use cases for smart accounts, empowering users with the ability to tailor authentication and recovery mechanisms to their specific needs. To unlock the potential for more convenient verification methods such as social login, we inevitably need to connect smart accounts and OpenID Connect(OIDC), given its status as the most widely accepted authentication protocol. In this EIP, we proposed a new model to use OIDC identity as the recovery mechanism for smart accounts. Users can link their accounts with OIDC identities and authorize an OIDC verifier to mutate their account state, such as resetting operators, by performing ZK verification against the linked OIDC identity. The corresponding interfaces are also defined in the spec.
+
+## Motivation
+
+Connecting OIDC identity and smart accounts has been a very interesting but challenging problem. The most common ways of tackling the issue is by utilizing Multi-Party Computation(MPC). However, the limitation of the MPC solution is obvious:
```

Comment:
Since ZKP and OpenID Connect is the central piece, it would be helpful you establish these two concepts as context properly, so I'd recommend sufficient introduction of them.

Reply:
It's hard to illustrate a specific ZKP proof structure since different ZKP algorithm may generate very different proofs. One of the goal of the design is to be ZK compatible but it can be used to support non-zk proofs as well. I will update the description to focus on the OpenID verification. How we do the OpenID verification is explained in the workflow and Rationale section but I can add some more context.